src/Symfony/Component/Serializer/Encoder/ChainDecoder.php

```diff
@@ -24,12 +24,17 @@
  */
 class ChainDecoder implements ContextAwareDecoderInterface
 {
-    private array $decoders = [];
+    /**
+     * @var array<string, array-key>
+     */
     private array $decoderByFormat = [];
 
-    public function __construct(array $decoders = [])
-    {
-        $this->decoders = $decoders;
+    /**
+     * @param array<DecoderInterface> $decoders
+     */
+    public function __construct(
+        private readonly array $decoders = []
```

Comment:
I think I've said this before, but strictly speaking this is a BC break: the `$decoders` property has no longer a default value (the default value is given to the parameter only).
This creates issues when overriding the constructor without calling the parent as the property becomes uninitialized: https://3v4l.org/DId0c

Is this something we consider too much of an edge case, or would this be a reason against transforming everything to CPP?

Reply:
This has already beaten us at least once so this is a valid concern to me.